jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcDataSet.java

```diff
@@ -197,8 +197,12 @@ public void close() {
         if (_closed) {
             return;
         }
+
+        FileHelper.safeClose(_resultSet);
+
         if (_jdbcDataContext != null) {
-            _jdbcDataContext.close(_connection, _resultSet, _statement);
+            FileHelper.safeClose(_statement);
```

Comment:
No, we can't close the statement unconditionally, it makes almost every single test fail. I'm not entirely sure why (didn't really pursue it, honestly).